python/pyarrow/_compute.pyx

```diff
@@ -899,26 +900,26 @@ cdef class _TakeOptions(FunctionOptions):
 
 
 class TakeOptions(_TakeOptions):
-    def __init__(self, *, boundscheck=True):
+    def __init__(self, boundscheck=True):
```

Comment:
I would keep this (it's to force `boundscheck` to be a keyword argument, and never positional argument)

(and the same for some cases below)

Reply:
I remove all the keyword arguments simply to get a code review like yours. How do we decide what options should be positional and which ones should be keyword-only?

Reply:
Ok, I started doing that (still WIP), but also noticed that in C++ FunctionOptions, there are inconsistencies between the number of constructors, the availability of a `Defaults()` method, and data members. I am looking into this (need to reach out to other developers) to check what changes we are allowed to make at this time.